### ISSUE TITLE
Overly restrictive host validation on the backend.

### ISSUE THREAD

**Issue by aalekhpatel07** [memgraph/memgraph-platform#52]
Hi there,
Thank you for creating and maintaining this amazing project.

I had a couple of questions/concerns that I was hoping to get some assistance with.

### Setting up Memgraph Lab with an existing Memgraph deployment.

I was wondering if I could use Lab to explore my already running `memgraph/memgraph` container. I tried running a `memgraph/memgraph-platform` image locally and pointing to a deployed memgraph instance (say, at `https://memgraph`) but I'm faced with a dubious error message:
```
Host must be a valid hostname or IP address.
```

I've tried specifying the following endpoint/port pairs but all cause the same error:
- `https://memgraph`:`7687`
- `memgraph`:`7687`
- `memgraph.<staging-host>`:`7687`

I can confirm that the deployment at `https://memgraph.<staging-host>:7687` is reachable from the Lab container as well as my local machine since I receive this response when reached from the browser:
```
�†±¢„codeÐMemgraph.ExecutionException‡messageÐWSomething went wrong while executing the query! Check the server logs for more details.��
```

Aside: I'm also curious to learn what led to the decision of bundling the UI tool (Lab) and the db (Memgraph) in a single docker image instead of allowing them to be deployed independently.

---

### Failing `git submodule update --init --recursive`

I'm not sure if the Lab repo has been moved somewhere else (since https://github.com/memgraph/lab is 404 for me) and `git submodule update --init --recursive` also fails with the following error message:
```
Cloning into '/home/aalekh/memgraph-platform/lab'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:memgraph/lab' into submodule path '/home/aalekh/memgraph-platform/lab' failed
Failed to clone 'lab'. Retry scheduled
Cloning into '/home/aalekh/memgraph-platform/mage'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:memgraph/mage.git' into submodule path '/home/aalekh/memgraph-platform/mage' failed
Failed to clone 'mage'. Retry scheduled
```

**Comment by aalekhpatel07** [comment]
## Update

So it turns out that my `<staging-host>` has a tld that has numeric characters, and sleuthing around in the Lab container's `lab/dist-backend` revealed that Memgraph Lab relies on [npm class-validator](https://github.com/typestack/class-validator/blob/develop/src/decorator/string/IsFQDN.ts) that delegates to [npm validator.js](https://github.com/validatorjs/validator.js/blob/master/src/lib/isFQDN.js).

This [test](https://github.com/validatorjs/validator.js/compare/master...aalekhpatel07:validator.js:master) that asserts `isFQDN` rejects hosts with numeric tld by default, which explains the error I'm facing. 

### Proposed fix

There are two ways to go about this:

- Pass `{allow_numeric_tld: true}` to the host validation in the backend code. (This fixes my issue but someone might encounter another case in the future where a legitimately valid host is rejected due to overly restrictive or poorly configured validation schemes).

OR

- Remove the host validation altogether since its unnecessarily redundant anyways. If the host is unreachable, we can let the lower level networking layers relay that to us.

### Next steps

Unfortunately, I can't contribute to Memgraph Lab since the repo is inaccessible to me. If I could gain access, I wouldn't mind issuing a PR or two to apply these fixes. 

Either way, lmk if there's anything I can do to help.

**Comment by tonilastre** [comment]
Hi @aalekhpatel07! This might be best Issue report where there is an issue report + finding a problem + proposing a solution :)

**Issue 1) Unable to build a Dockerimage**
Yep, you are right, Lab is not public - yet. Lab uses a lot of other internal libraries and products, so open-sourcing Lab would require open-sourcing all of them - even though it would open up a contribution ability, it would slow us down a lot due to the handling of such matters. This is just a decision for now, who knows what the future holds! Maybe even Lab as a pluggable system for custom plugins because we are open-sourcing parts of Lab every few months (e.g. http://github.com/memgraph/orb).

**Issue 2) Host must be a valid hostname or IP address.**
You found an issue and a solution. It will be fixed in the next Lab release. In the meantime, you can patch it on your end. Head out to `/lab/dist-backend/utils/validation.js` and add `return true` to disable validation overall:

```
            ...
            validator: {
                validate(value, _) {
                    return true; // add this line here
                    if (typeof value !== 'string') {
                        return false;
                    }
                    ...
```

You can then `docker commit` or `docker save` your fix.

**Comment by tonilastre** [comment]
Hi @aalekhpatel07, just wanna update you that this has been fixed in Lab v2.7.0: https://memgraph.com/docs/memgraph-lab/changelog#lab-v270---jun-28-2023 - there is a line in bug fixes:

```
...
By fixing a bug, you can now successfully connect to Memgraph using a hostname that contains numbers in the top-level domain.
...
```

just make sure that `memgraph-platform` image that you are running has `lab-2.7.0`, `lab-2.8.0`, or `lab-2.8.1` in the tag name. Currently, the latest Lab version is v2.8.1 with even more features and improvements. Thanks again for the report! :)

**Comment by aalekhpatel07** [comment]
Awesome! Thanks for the update. 🚀
